scripts/build.py

```diff
@@ -684,26 +823,137 @@ def get_recently_changed_filenames(dev_dir, out_dir):
     # the filenames and their extensions,
     # e.g base.240933e7564bd72a4dde42ee23260c5f.html
     # If a file gets edited, a different MD5 hash is generated.
-    file_hashes = get_file_hashes(dev_dir)
     recently_changed_filenames = []
     # Currently, we do not hash Python files and HTML files are always re-built.
     FILE_EXTENSIONS_NOT_TO_TRACK = ('.html', '.py',)
-    print "Comparing file hashes of %s and %s" % (dev_dir, out_dir)
-    for file_name, md5_hash in file_hashes.iteritems():
+    for filename, md5_hash in source_dir_hashes.iteritems():
         # Ignore files with certain extensions.
-        if any(file_name.endswith(p) for p in FILE_EXTENSIONS_NOT_TO_TRACK):
+        if any(filename.endswith(p) for p in FILE_EXTENSIONS_NOT_TO_TRACK):
             continue
         final_filepath = _insert_hash(
-            os.path.join(out_dir, file_name), md5_hash)
+            os.path.join(out_dir, filename), md5_hash)
         if not os.path.isfile(final_filepath):
             # Filename with provided hash cannot be found, this file has been
             # recently changed or created since last build.
-            recently_changed_filenames.append(file_name)
-    print ('The following files will be rebuilt due to recent changes: %s' %
-           recently_changed_filenames)
+            recently_changed_filenames.append(filename)
+    if recently_changed_filenames:
+        print ('The following files will be rebuilt due to recent changes: %s' %
+               recently_changed_filenames)
     return recently_changed_filenames
 
 
+def generate_build_tasks_to_build_directory(dirnames_dict, file_hashes):
+    """This function queues up build tasks to build all files in source
+    directory if there is no existing staging directory. Otherwise, selectively
+    queue up build tasks to build recently changed files.
+
+    Args:
+        dirnames_dict: dict(str, str, str). A dict containing directory paths to
```

Comment:
dict(str, str).

In the description, specify keys/values fully. E.g.:

```
  This dict should contain three keys, with the corresponding values as follows:
  - 'dev_dir': The directory that contains ...
  - ...
  - ...
```

Reply:
Clarified docstring and fixed dict type.